pytools/__init__.py

```diff
@@ -2041,12 +2074,12 @@ def typedump(val, max_seq=5, special_handlers=None):
         try:
             if len(val) > max_seq:
                 return "{}({},...)".format(
-                        type(val).__name__,
+                        objname(val),
                         ",".join(typedump(x, max_seq, special_handlers)
                             for x in val[:max_seq]))
             else:
                 return "{}({})".format(
-                        type(val).__name__,
+                        objname(val),
                         ",".join(typedump(x, max_seq, special_handlers)
                             for x in val))
         except TypeError:
```

Comment:
This could probably be changed to `objname(val)` as well, but I couldn't come up with an example that hits this exception here.

Reply:
changed in [7c43184](https://github.com/inducer/pytools/pull/174/commits/7c43184dd3d0d28f6956e94616014c5c405fb00f)